### DISCUSSION TITLE
Materialize view purging or lifespan

### DISCUSSION THREAD

**Discussion by mans2singh** [risingwavelabs/risingwave#9541] [Q&A] [answered]
Hi:

Is there any way to run a materialize query for a certain time span ?

From what I understand, a materialized view runs forever.  But is there any ability to apply some time constraints on the view and then expire the materialized view ?

Thanks

**Comment by lmatz** [comment, marked as answer]
Hi @mans2singh,

by `expiring the materialized view`, do you want to see that the data being materialized in the view is always restricted to a recent period of time, or do you want to see that this materialized view gets `dropped` after XX minutes?

The first one is supported by temporal filters: https://www.risingwave.dev/docs/current/sql-pattern-temporal-filters/
The examples in the doc also applies to `create materialized view XXX as select ...;`

We don't support the second one right now. I would recommend doing it out of RW, e.g. from a client to the database. Set a cron job etc.

**Reply by yingjunwu** [reply]
Agreed with @lmatz - if you are asking for data retention, then that's what RisingWave can offer; if you want to drop materialized views after certain time period, then I would recommend using external program to do that. RisingWave as a database is not supposed to create/drop table/view on its own.

**Reply by mans2singh** [reply]
Thanks for your help @lmatz @yingjunwu .
